native/thrift.c

```diff
@@ -1061,10 +1061,6 @@ uint64_t j2t_fsm_exec(J2TStateMachine *self, GoSlice *buf, const GoString *src,
             xprintf("[J2T_VAL] null\n");
             long s = *p;
             J2T_XERR(advance_dword(src, p, 1, *p - 1, VS_NULL), s);
-            if (IS_STATE_FIELD(st) && unlikely(vt->ex.ef.f->required == REQ_REQUIRED))
```

Comment:
It's used for check if it is a required field in past, but since very early all requireness checking has been handle over to `j2t_write_unset_fields()`. So it is unnecessary here, I just forget to delete it

Reply:
@Marina-Sakai